### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by balbirthomas**
Refactored MongoDB charm. Individual commit messages explain motivation for each refactoring.

### CODE REVIEW

File: src/charm.py [hunk 1]
```diff
@@ -184,73 +151,72 @@ def reconfigure(self, event):
...
+            replica_set_uri=f"{self.replica_set_uri}?replicaSet={self.replica_set_name}",
         )
 
+    @property
+    def replica_set_uri(self):
+        uri = "mongodb://"
+        for i, host in enumerate(self.cluster_hosts):
+            if i:
+                uri += ","
+            uri += f"{host}:{self.port}"
```
Comment (Abuelodelanada) [inline src/charm.py:191]: This line is not Python 3.5 compatible. 
What about this:

```python
"{host}:{self.port}".format(host, self.port)
```

File: src/charm.py [hunk 2]
```diff
@@ -184,73 +151,72 @@ def reconfigure(self, event):
...
+        uri += "/"
+        return uri
+
+    @property
+    def standalone_uri(self):
+        return f"mongodb://{self.model.app.name}:{self.port}/"
```
Comment (Abuelodelanada) [inline src/charm.py:197]: This line is not Python 3.5 compatible. 
What about this:

```python
"mongodb://{self.model.app.name}:{self.port}/".format(self.model.app.name, self.port)
```

File: src/charm.py [hunk 3]
```diff
@@ -184,73 +151,72 @@ def reconfigure(self, event):
...
-            if config.get(setting) is None:
-                problem = f"missing config {setting}"
-                problems.append(problem)
-        if not self.standalone:
-            for setting in REQUIRED_SETTINGS_NOT_STANDALONE:
-                if not config.get(setting):
-                    problem = f"missing config {setting}"
-                    problems.append(problem)
+    def _get_unit_hostname(self, _id: int) -> str:
+        return f"{self.model.app.name}-{_id}.{self.model.app.name}-endpoints"
```
Comment (Abuelodelanada) [inline src/charm.py:212]: This line is not Python 3.5 compatible. 
What about this:

```python
"{self.model.app.name}-{_id}.{self.model.app.name}-endpoints".format(self.model.app.name, _id, self.model.app.name)
```

**Review by Abuelodelanada — APPROVED** [review]

**Comment by Abuelodelanada** [comment]
@balbirthomas One general comment: May be we can have "free" CI if we copy this file to the repo:
https://github.com/canonical/graylog-operator/blob/main/.github/workflows/python-package.yml

File: src/charm.py (hunk 1 above)
Comment (balbirthomas) [reply src/charm.py:191]: Thank you will fix.

File: src/charm.py (hunk 2 above)
Comment (balbirthomas) [reply src/charm.py:197]: Thanks. Will fix.

File: src/charm.py (hunk 3 above)
Comment (balbirthomas) [reply src/charm.py:212]: Thanks again. I feel silly now for not having paid attention to these details.

File: metadata.yaml [hunk 4]
```diff
@@ -1,7 +1,9 @@
 name: mongodb
-summary: A MongoDB K8s charm
+summary: A MongoDB operator charm for Kubernets
```
Comment (justinmclark) [inline metadata.yaml:2]: minor typo in Kubernetes

**Review by justinmclark — APPROVED** [review]
Just a minor typo. Looks good to me and works well when I deployed it for Graylog.

File: metadata.yaml (hunk 4 above)
Comment (balbirthomas) [reply metadata.yaml:2]: thanks. will fix.

**Comment by balbirthomas** [comment]
> @balbirthomas One general comment: May be we can have "free" CI if we copy this file to the repo:
> https://github.com/canonical/graylog-operator/blob/main/.github/workflows/python-package.yml

Thank you very much for this. I will add it along with my next pull request that will include some meaningful unit tests.

**Pull request merged**